src/init.cpp

```diff
@@ -376,7 +374,7 @@ static void HandleSIGHUP(int)
 #else
 static BOOL WINAPI consoleCtrlHandler(DWORD dwCtrlType)
 {
-    StartShutdown();
+    Assert(kernel::g_context)->interrupt();
```

Comment:
re: https://github.com/bitcoin/bitcoin/pull/28051#discussion_r1257291194

> I was looking at the initialization order here and noticed that I might have introduced more unsafe behavior here in [e2d680a](https://github.com/bitcoin/bitcoin/commit/e2d680a32d757de0ef8eb836047a0daa1d82e3c4). These handlers are registered in `AppInitBasicSetup`. But if you look in bitcoind.cpp, the kernel context is created after `AppInitBasicSetup` is called.

Yes this is now fixed by having ctrl-c and signal handler use a g_shutdown global instead of using the kernel context.